it/src/main/java/com/google/cloud/teleport/it/jdbc/AbstractJDBCResourceManager.java

```diff
@@ -0,0 +1,363 @@
+   *
+   * @param tableName The name of the table to insert the given row into.
+   * @param row An Object array containing the row values.
+   * @throws JDBCResourceManagerException if method is called after resources have been cleaned up,
+   *     if the table does not exist or if there is an Exception when attempting to insert the rows.
+   * @return true, if the rows were successfully written
+   */
+  public boolean write(String tableName, Integer id, Object... row) {
+    return write(tableName, ImmutableMap.of(id, ImmutableList.of(row)), ImmutableList.of());
+  }
+
+  @Override
+  public boolean write(String tableName, Map<Integer, List<Object>> rows) {
+    return write(tableName, rows, ImmutableList.of());
+  }
+
+  /**
+   * Writes the given mapped rows into the specified columns. This method requires {@link
+   * JDBCResourceManager#createTable(String, JDBCSchema)} to be called for the target table
+   * beforehand.
+   *
+   * <p>The rows map must use the row id as the key, and the values will be inserted into the columns at the row with
+   * that id. i.e. {0: [val1, val2, ...], 1: [val1, val2, ...], ...}
+   *
+   * @param tableName The name of the table to insert the given rows into.
+   * @param rows A map representing the rows to be inserted into the table.
+   * @param columns A list of columns to write the mapped values to.
+   * @throws JDBCResourceManagerException if method is called after resources have been cleaned up,
+   *     if the manager object has no dataset, if the table does not exist or if there is an
+   *     Exception when attempting to insert the rows.
+   */
+  public boolean write(String tableName, Map<Integer, List<Object>> rows, List<String> columns) {
+    LOG.info("Attempting to write {} rows to {}.{}.", rows.size(), databaseName, tableName);
+
+    if (rows.size() == 0) {
+      return false;
+    }
+
+    StringBuilder sql = new StringBuilder();
+    try (Connection con = driver.getConnection(getUri(), username, password)) {
+      Statement stmt = con.createStatement();
+      sql.append("INSERT INTO ").append(tableName);
+      if (columns.size() > 0) {
+        sql.append("(").append(String.join(",", columns)).append(")");
+      }
+      sql.append(" VALUES ");
+
+      List<String> rowsToInsert = new ArrayList<>();
+      for (Integer rowId : rows.keySet()) {
+        StringBuilder valueList = new StringBuilder("(");
+
+        List<String> rowToInsert = new ArrayList<>();
+        rowToInsert.add(String.valueOf(rowId));
+        for (Object value : rows.get(rowId)) {
+          if (!NumberUtils.isCreatable(value.toString())) {
+            rowToInsert.add("'" + value + "'");
+          } else {
+            rowToInsert.add(String.valueOf(value));
+          }
+        }
+        valueList.append(String.join(",", rowToInsert)).append(")");
+
+        rowsToInsert.add(valueList.toString());
+      }
+      stmt.executeUpdate(sql.append(String.join(",", rowsToInsert)).toString());
```

Comment:
nit: maybe print the SQL statement being executed for better?

Reply:
I added a log statement. Let me know if you think we should use a different log level or if the message should be changed.